transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/TransportRuntime.java

```diff
@@ -109,6 +110,16 @@ public TransportFactory newFactory(String backendName) {
         TransportContext.builder().setBackendName(backendName).build(), this);
   }
 
+  /** Returns a {@link TransportFactory} for a given {@code backendName}. */
+  public TransportFactory newFactory(Destination destination) {
```

Comment:
We also want to deprecate the old overload, is it a good time to do now? or can you do this in a follow up PR?

Reply:
By deprecate, if you mean an active deprecation of moving clients over to the new API, I can do that separately.